src/once.rs

```diff
@@ -50,12 +50,73 @@ impl<T: fmt::Debug, R> fmt::Debug for Once<T, R> {
 unsafe impl<T: Send + Sync, R> Sync for Once<T, R> {}
 unsafe impl<T: Send, R> Send for Once<T, R> {}
 
-// Four states that a Once can be in, encoded into the lower bits of `status` in
-// the Once structure.
-const INCOMPLETE: usize = 0x0;
-const RUNNING: usize = 0x1;
-const COMPLETE: usize = 0x2;
-const PANICKED: usize = 0x3;
+mod status {
+    use super::*;
+
+    // SAFETY: This structure has an invariant, namely that the inner atomic u8 must *always* have
+    // a value for which there exists a valid Status. This means that users of this API must only
+    // be allowed to load and store `Status`es.
+    pub struct AtomicStatus(AtomicU8);
```

Comment:
Not really needed yet, but I think good to be future-proof like this?
```suggestion
    #[repr(transparent)]
    pub struct AtomicStatus(AtomicU8);
```

Reply:
Done.